ts/store/reducers/crossSessions.ts

```diff
@@ -26,13 +38,25 @@ const reducer = (
   state: CrossSessionsState = INITIAL_STATE,
   action: Action
 ): CrossSessionsState => {
-  if (isActionOf(setProfileHashedFiscalCode, action)) {
-    return {
-      ...state,
-      hashedFiscalCode: hash(action.payload)
-    };
+  switch (action.type) {
```

Comment:
Was the refactoring of this reducer needed?

I just made a check by only using the `isDifferentFiscalCodeSelector` on the startup saga and i got the same effect expected by the story

in this way we have a sort of redundant information about cross session login: we store the hashed fiscal code, among the information that is the same or different identity logged in.

imho we can avoid this refactoring and move on by only comparing the current fiscal code with the hashed one previously stored, the information can be found by executing a single selector.

the saga would look like follows:

```typescript
    if (isNone(maybeStoredPin)) {
      storedPin = yield call(checkConfiguredPinSaga);
    } else {
      const isDifferentProfile: CrossSessionProfileIdentity = yield select(
        isDifferentFiscalCodeSelector,
        userProfile.fiscal_code
      );

      if (isDifferentProfile) {
        yield call(deletePin);
        storedPin = yield call(checkConfiguredPinSaga);
      } else {
        storedPin = maybeStoredPin.value;
      }
```

I think it's a clearer solution.

Reply:
Sorry, didn't really grasp that we could easily access the current fiscal code by `userProfile.fiscal_code`

Nevertheless, storing a `isDifferentProfile` flag in the store could be useful in the future, in case we need to check this in other sagas with the current fiscal code not being ready to access through a selector.

Are you sure you want to drop this additional step? In that case, I'll double check your solution and go on.

Reply:
In case we are going to need it in future we will evaluate to add that. We can always easily access the user fiscal code by currently logged user information without the need of another selector. 😉

Reply:
I've tried the simplification you suggested but still get an unexpected result from `isDifferentFiscalCodeSelector`, because the `CROSS_SESSION_SET_HASHED_CF` action happens before this call:

```
yield select(
        isDifferentFiscalCodeSelector,
        userProfile.fiscal_code
 );
 ```
 The action is dispatched upon calling `loadProfile` which dispatches `loadProfileSuccess` which triggers `checkLoadedProfile`
 
 So I guess we either change the flow or store either both the "old" fiscal code and the new one or a flag as i did.

Reply:
Commit d572a8e contains the code related to this attempt.